flink-runtime/src/main/java/org/apache/flink/runtime/clusterframework/types/ResourceProfile.java

```diff
@@ -640,15 +642,18 @@ public Builder setNetworkMemoryMB(int networkMemoryMB) {
             return this;
         }
 
-        public Builder addExtendedResource(String name, Resource extendedResource) {
-            this.extendedResources.put(name, extendedResource);
+        // Add the given extended resource, the old value with the same resource name will be
+        // override if present.
+        public Builder setExtendedResource(Resource extendedResource) {
+            this.extendedResources.put(extendedResource.getName(), extendedResource);
             return this;
         }
 
-        public Builder addExtendedResources(Map<String, Resource> extendedResources) {
-            if (extendedResources != null) {
-                this.extendedResources.putAll(extendedResources);
-            }
+        // Add the given extended resources, this will override all the previous records.
+        public Builder setExtendedResources(Collection<Resource> extendedResources) {
+            this.extendedResources =
+                    extendedResources.stream()
+                            .collect(Collectors.toMap(Resource::getName, Function.identity()));
```

Comment:
JavaDoc style is preferred. It hints the compiler/IDE that there's a binding between the docs and the following class/field/method.

Reply:
Same for `ResourceSpec.Builder` and `WorkerResourceSpec.Builder`.